operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Metrics.java

```diff
@@ -0,0 +1,152 @@
+package io.javaoperatorsdk.operator;
+
+import io.fabric8.kubernetes.client.CustomResource;
+import io.javaoperatorsdk.operator.api.Context;
+import io.javaoperatorsdk.operator.api.DeleteControl;
+import io.javaoperatorsdk.operator.api.ResourceController;
+import io.javaoperatorsdk.operator.api.UpdateControl;
+import io.javaoperatorsdk.operator.api.config.ControllerConfiguration;
+import io.micrometer.core.instrument.*;
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
+import io.micrometer.core.instrument.distribution.pause.PauseDetector;
+import io.micrometer.core.instrument.noop.*;
+import java.util.concurrent.TimeUnit;
+import java.util.function.ToDoubleFunction;
+import java.util.function.ToLongFunction;
+
+public class Metrics {
+  public static final Metrics NOOP = new Metrics(new NoopMeterRegistry(Clock.SYSTEM));
+  private final MeterRegistry registry;
+
+  public Metrics(MeterRegistry registry) {
+    this.registry = registry;
+  }
+
+  public <R extends CustomResource> UpdateControl<R> timeControllerCreateOrUpdate(
+      ResourceController<R> controller,
+      ControllerConfiguration<R> configuration,
+      R resource,
+      Context<R> context) {
+    final var name = configuration.getName();
+    final var timer =
+        registry.timer("operator.sdk.controllers.execution.createOrUpdate", "controller", name);
+    try {
+      final var result = timer.record(() -> controller.createOrUpdateResource(resource, context));
+      String successType = "cr";
+      if (result.isUpdateStatusSubResource()) {
+        successType = "status";
+      }
+      if (result.isUpdateCustomResourceAndStatusSubResource()) {
+        successType = "both";
+      }
+      registry
+          .counter(
+              "operator.sdk.controllers.execution.success", "controller", name, "type", successType)
+          .increment();
+      return result;
+    } catch (Exception e) {
+      registry
+          .counter(
+              "operator.sdk.controllers.execution.failure",
+              "controller",
+              name,
+              "exception",
+              e.getClass().getSimpleName())
+          .increment();
+      throw e;
+    }
+  }
+
+  public DeleteControl timeControllerDelete(
+      ResourceController controller,
+      ControllerConfiguration configuration,
+      CustomResource resource,
+      Context context) {
+    final var name = configuration.getName();
+    final var timer =
+        registry.timer("operator.sdk.controllers.execution.delete", "controller", name);
+    try {
+      final var result = timer.record(() -> controller.deleteResource(resource, context));
+      String successType = "cr";
```

Comment:
Do you really think that this success type makes sense for the delete operation?

Reply:
```
      String successType = "notDelete";
      if (result == DeleteControl.DEFAULT_DELETE) {
        successType = "delete";
      }
 ```

I modified like this.